polaris-core/src/main/java/org/apache/polaris/core/persistence/cache/InMemoryEntityCache.java

```diff
@@ -76,15 +77,22 @@ public InMemoryEntityCache(@Nonnull PolarisMetaStoreManager polarisMetaStoreMana
         };
 
     long weigherTarget =
-        PolarisConfiguration.loadConfig(FeatureConfiguration.ENTITY_CACHE_WEIGHER_TARGET);
+        polarisCallContext
+            .getConfigurationStore()
```

Comment:
I looked through the code and as far as I can see all instances of `PolarisConfigurationStore` happen to be application-scoped. Could we avoid requiring a "call" (i.e. request) context for getting `PolarisConfigurationStore`?

If injection through CDI seems too intrusive for this PR, could we instead use `PolarisConfigurationStore` as the new parameter (instead of `PolarisCallContext`)?

Reply:
but TBH, I'd really like to use CDI (with custom producers) for injection rather than relying on factories in the main call path. In other words, I'd like the main body of code to receive dependent objects through injection and factories / producers to be removed from the main call paths and be engaged by CDI when request-handlers are constructed.

All-in-all, the intention behind this PR seems to be pointing in that direction, so why not do the CDI-based refactoring now and then perhaps the config loading issue is solved automatically?